### ISSUE TITLE
Making channels_last is a bug?

### ISSUE THREAD

**Issue by RicoJia** [milesial/Pytorch-UNet#514]
I understand that this is for mixed precision training. 
```python
    model = model.to(memory_format=torch.channels_last)
```

However, in your

```python
    def forward(self, x1, x2):
        x1 = self.up(x1)
        # input is CHW
        diffY = x2.size()[2] - x1.size()[2]
        diffX = x2.size()[3] - x1.size()[3]

        x1 = F.pad(x1, [diffX // 2, diffX - diffX // 2,
                        diffY // 2, diffY - diffY // 2])
        # if you have padding issues, see
        # https://github.com/HaiyongJiang/U-Net-Pytorch-Unstructured-Buggy/commit/0e854509c2cea854e247a9c615f175f76fbb2e3a
        # https://github.com/xiaopeng-liao/Pytorch-UNet/commit/8ebac70e633bac59fc22bb5195e513d5832fb3bd
        x = torch.cat([x2, x1], dim=1)
        return self.conv(x)
```

This is assuming CHW and hardcoded padding based on the assumption. This could cause issues?
